sdks/java/extensions/sketching/src/main/java/org/apache/beam/sdk/extensions/sketching/frequency/SketchFrequencies.java

```diff
@@ -0,0 +1,351 @@
+   *
+   * <p>The {@code seed} parameters will be used to randomly generate different hash functions.
+   * Thus, the result can be different for the same stream in different seeds are used.
+   * The {@code seed} parameter will be used to generate a and b for each hash function.
+   * <br>The Count-min sketch size is constant through the process so the memory use is fixed.
+   * However, the dimensions are directly linked to the accuracy.
+   * <br>By default, the relative error is set to 1% with 1% probability that the estimation
+   * breaks this limit.
+   * <br>also keep in mind that this algorithm works well on highly skewed data but gives poor
+   * results if the elements are evenly distributed.
+   *
+   * <p>See {@link CountMinSketchFn#withAccuracy(double, double)} in order to tune the parameters.
+   * <br>Also see {@link CountMinSketchFn} for more details about the algorithm's principle.
+   *
+   * <p>Example of use:
+   * <pre> {@code
+   * PCollection<KV<Integer, String>> pc = ...;
+   * PCollection<KV<Integer, CountMinSketch>> countMinSketch =
+   *     pc.apply(SketchFrequencies.<Integer, String>perKey(1234));
+   * } </pre>
+   *
+   * @param seed        the seed used for generating different hash functions
+   * @param <K>         the type of the keys in the input and output {@code PCollection}s
+   */
+  public static <K> Combine.PerKey<K, String, CountMinSketch> perKey(int seed) {
+    return Combine.<K, String, CountMinSketch>perKey(CountMinSketchFn
+            .create(seed).withAccuracy(0.001, 0.99));
+  }
+
+  /**
+   * A {@code Combine.CombineFn} that computes the {@link CountMinSketch} Structure
+   * of an {@code Iterable} of Strings, useful as an argument to {@link Combine#globally} or
+   * {@link Combine#perKey}.
+   *
+   * <p>When an element is added to the Count-min sketch, it is mapped to one column in each
+   * row using different hash functions, and a counter is updated in each column.
+   * <br>Collisions will happen as far as the number of distinct elements in the stream is greater
+   * than the width of the sketch. Each counter might be associated to many items, so the frequency
+   * of an element is always overestimated. On average the relative error on a counter is bounded,
+   * but some counters can be very inaccurate.
+   * <br>That's why different hash functions are used to map the same element to different
+   * counters. Thus, the overestimation for each counter will differ as there will be different
+   * collisions, and one will probably be less inaccurate than the average.
+   *
+   * <p>Both the average relative error and the probability to have an estimation overcoming this
+   * error can be computed by knowing the dimensions of the sketch, and vice-versa.
+   * Thus, for Count-min sketch with 10 000 columns and 7 rows, the relative error should not be no
+   * more than 0.02% in 99% of the cases.
+   *
+   */
+  static class CountMinSketchFn
```

Comment:
<!--new_thread; commit:6bf51fdf6afe6f389415792f335dd08adf9fa85a; resolved:0-->
You could support all types instead of just strings by encoding each value to a byte[] and then converting that byte array to a base64 string or something equivalent.

You just want to make sure that the coder is deterministic.